### DISCUSSION TITLE
Error executing Lua callback when triggering CodeCompanionToggle

### DISCUSSION THREAD

**Discussion by englut** [olimorris/codecompanion.nvim#30] [Q&A] [answered]
Hey y'all. I'm not sure what I'm doing wrong, hoping maybe someone can point me in the right direction. I've been using Code Companion on nvim and it worked fine. I was initiating everything via my `init.vim`. Using `Plug` and all that. Everything worked great. But I'm currently refactoring my nvim config to be pure `init.lua` instead (using lazy.nvim). 

My issue is that when I trigger `CodeCompanionToggle`, I'm seeing some treesitter errors that I've never seen before. I definitely have markdown enabled for treesitter too.

<img width="1117" alt="Screenshot 2024-03-22 at 23 59 45" src="https://github.com/olimorris/codecompanion.nvim/assets/15840269/1ab97002-99cc-4dc1-aeb5-3bbcb8f0619e">

Does anyone know where I can read more logs or try and understand what might be wrong with my setup?
I've attempted to isolate the plugin (commenting out all of my config, except the plugin, it's dependencies and the key bindings) and the error still happens.

**Comment by olimorris** [comment, marked as answer]
Running `:checkhealth codecompanion` will show you where logs are but I'm not sure I've built anything in for checking Treesitter parsers.

Try `:TSInstall markdown` too.

**Reply by englut** [reply]
Oh dear. Looks like I had set up treesitter incorrectly! I've fixed it and now Code Companion works again! I should have known to test treesitter's installations 🥲  Thank you for pointing me in the right direction 🙏

**Reply by olimorris** [reply]
Glad it's sorted! Also, let me know any feedback or ideas you have

**Comment by TheCedarPrince** [comment]
Hey @qauff and @olimorris , I am seeing this error again on neovim 0.9.5 and the latest version of code companion. Any idea what is happening here? I did definitely check the health of codecompanion and everything is fine as well as running `TSInstall markdown`. Still running into errors.

![image](https://github.com/olimorris/codecompanion.nvim/assets/29561456/acff87eb-e7b5-4dd8-8e24-c1de0447397e)

**Reply by olimorris** [reply]
I'd suggest opening up an issue for this and trying to recreate it with a `minimal.lua` file. I'll gladly look into it.

**Reply by TheCedarPrince** [reply]
Can do @olimorris ! :smile:
